app/controllers/reservations_controller.rb

```diff
@@ -1,35 +1,54 @@
 class ReservationsController < ApplicationController
   include Pundit::Authorization
 
-  before_action :set_reservation, only: %i[show edit update destroy]
+  before_action :set_reservation, only: %i[show edit update destroy cancel_reservation confirm_reservation]
   before_action :authenticate_user!
 
   def find_reservations_by_user
-    @reservations = Reservation.where(user_id: find_user_by_email(params[:email]))
-                               .joins(:seance).includes([:user, { seance: [:movie] }])
-                               .order(start_time: :desc)
-    render template: "reservations/index"
+    authorize Reservation
+    index(find_user_by_email(params[:email]))
   end
 
   def find_reservations_by_seance
-    @reservations = Reservation.where(seance_id: params[:seance_id]).includes([:user, { seance: [:movie] }])
+    authorize Reservation
+    @reservations = Reservation.where(seance_id: params[:seance_id])
+                               .includes([:user, { seance: [:movie] }])
     render template: "reservations/index"
   end
 
-  def index
-    @reservations = Reservation.where(user_id: current_user.id).includes([:user, { seance: [:movie] }])
+  def cancel_reservation
+    authorize @reservation
+    update(status: :canceled)
+  end
+
+  def confirm_reservation
+    authorize @reservation
+    update(status: :confirmed)
+  end
+
+  def index(user_id = current_user.id)
+    authorize Reservation
+    @reservations = Reservation.where(user_id:)
```

Comment:
It looks a little bit unnatural to me. The controller methods usually are called without passing arguments, so the current version could lead to confuse other readers
```suggestion
  def index
    authorize Reservation
    @reservations = Reservation.where(user_id: current_user.id)
```

Reply:
`index` did the same thing as method `find_by_user` only with different id, so I tried to use index in the second method but now I just moved the query to another method so the index stays without any arguments and they both can use it :)